oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/query/IMetricsQueryDAO.java

```diff
@@ -76,5 +87,54 @@ public static IntValues sortValues(IntValues origin, List<String> expectedOrder,
             }
             return origin;
         }
+
+        /**
+         * Compose the multiple metric result based on conditions.
+         */
+        public static List<MetricsValues> composeLabelValue(final MetricsCondition condition,
+            final List<String> labels,
+            final List<String> ids,
+            final Map<String, DataTable> idMap) {
+            List<String> allLabels;
+            if (Objects.isNull(labels) || labels.size() < 1) {
+                allLabels = idMap.values().stream()
+                    .flatMap(dataTable -> dataTable.keys().stream())
+                    .distinct().collect(Collectors.toList());
+            } else {
+                allLabels = labels;
+            }
+            final int defaultValue = ValueColumnMetadata.INSTANCE.getDefaultValue(condition.getName());
+            List<LabeledValue> labeledValues = Sets.newTreeSet(allLabels).stream()
+                .flatMap(label -> ids.stream().map(id ->
+                    new LabeledValue(label, id, Optional.ofNullable(
+                        idMap.getOrDefault(id, new DataTable()).get(label)).orElse((long) defaultValue))))
+                .collect(toList());
+            MetricsValues current = new MetricsValues();
+            List<MetricsValues> result = new ArrayList<>();
+            for (LabeledValue each : labeledValues) {
+                if (!Strings.isNullOrEmpty(current.getLabel()) && each.label.equals(current.getLabel())) {
```

Comment:
You are using `each.label.equals`, then `!Strings.isNullOrEmpty(current.getLabel())` isn't required, right?
```suggestion
                if (each.label.equals(current.getLabel())) {
```
Or you need to check `each.label != null`, from my understanding.

Reply:
`Objects.equals` is a better choice.